src/cli/cli.cpp

```diff
@@ -602,6 +612,259 @@ void Interpreter::ProcessChildTimeout(int argc, char *argv[])
+        otCoapHeader responseHeader;
+        otMessage *responseMessage;
+        otCoapCode responseCode = kCoapResponseValid;
+        char responseContent = '0';
+
+        if (otCoapHeaderGetCode(aHeader) == kCoapRequestGet)
+        {
+            responseCode = kCoapResponseContent;
+        }
+        otCoapHeaderInit(&responseHeader, kCoapTypeAcknowledgment, responseCode);
+        otCoapHeaderSetMessageId(&responseHeader, otCoapHeaderGetMessageId(aHeader));
+        otCoapHeaderSetToken(&responseHeader, otCoapHeaderGetToken(aHeader), otCoapHeaderGetTokenLength(aHeader));
+        if (otCoapHeaderGetCode(aHeader) == kCoapRequestGet)
+        {
+            otCoapHeaderSetPayloadMarker(&responseHeader);
+        }
+        responseMessage = otCoapNewMessage(mInstance, &responseHeader);
+        if (otCoapHeaderGetCode(aHeader) == kCoapRequestGet)
+        {
+            otMessageAppend(responseMessage, &responseContent, sizeof(responseContent));
+        }
+
+        error = otCoapSendResponse(mInstance, responseMessage, aMessageInfo);
+
+        if (error != kThreadError_None)
+        {
+            sServer->OutputFormat("Cannot send CoAP response message: Error %d\r\n", error);
+            otMessageFree(responseMessage);
+        }
+        else
+        {
+            sServer->OutputFormat("CoAP response sent successfully!\r\n");
+        }
+    }
+
+    (void)aHeader;
+    (void)aMessage;
+    (void)aMessageInfo;
+}
+
+void Interpreter::ProcessCoapClient(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+    otMessage *message;
+    otMessageInfo messageInfo;
+    otCoapHeader header;
+    const otNetifAddress *address;
+
+    // Default parameters
+    char coapUri[] = "test";
+    char coapContent[20] = "";
+    otCoapType coapType = kCoapTypeNonConfirmable;
+    otCoapCode coapCode = kCoapRequestGet;
+    otIp6Address coapDestinationIp;
+    otIp6AddressFromString("FF01::1", &coapDestinationIp); // Broadcast
+
+    VerifyOrExit(argc > 0, error = kThreadError_InvalidArgs);
+
+    // CoAP-Code
+    if (strcmp(argv[0], "GET") == 0)
+    {
+        coapCode = kCoapRequestGet;
+    }
+    else if (strcmp(argv[0], "POST") == 0)
+    {
+        coapCode = kCoapRequestPost;
+    }
+    else if (strcmp(argv[0], "PUT") == 0)
+    {
+        coapCode = kCoapRequestPut;
+    }
+    else if (strcmp(argv[0], "DELETE") == 0)
+    {
+        coapCode = kCoapRequestDelete;
+    }
+    else
+    {
+        ExitNow(error = kThreadError_Parse);
+    }
+
+    // Destination IPv6 address
+    if (argv[1])
+    {
+        SuccessOrExit(error = otIp6AddressFromString(argv[1], &coapDestinationIp));
+    }
+
+    // CoAP-URI
+    if (argv[2])
+    {
+        strcpy(coapUri, argv[2]);
+    }
+
+    // CoAP-Type
+    if (strcmp(argv[3], "Confirmable") == 0)
+    {
+        coapType = kCoapTypeConfirmable;
+    }
+
+    // Content
+    if (argv[4])
+    {
+        strcpy(coapContent, argv[4]);
```

Comment:
If coapContent has fixed size then you need a check if` argv[4] `does not exceed it.

But anyhow i think you can remove temporary `coapContent` buffer and use `otMessageAppend ` that you have in line 816 to append string directly to the message buffer.